File: Ayaan/leetcode/L94.py
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        def inorder(root, inorderList):
            if not root:
                return root
            
            if root.left:
                inorder(root.left, inorderList)
            inorderList.append(root.val)
            if root.right:
                inorder(root.right, inorderList)
            return inorderList

        inorderList = []
        return inorder(root, inorderList)

File: Ayaan/leetcode/L94.java
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> inorderList = new ArrayList<Integer>();
        return inorder(root, inorderList);
    }

    public List<Integer> inorder(TreeNode node, List<Integer> inorderList){
        if(node == null){
            return inorderList;
        }

        TreeNode left = node.left;

        if(left != null){
            inorder(left, inorderList);
        }

        inorderList.add(node.val);

        TreeNode right = node.right;

        if(right != null){
            inorder(right, inorderList);
        }

        return inorderList;
    }
}